application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/config/connection/AgentConnectionConfig.java

```diff
@@ -0,0 +1,315 @@
+        }
+    }
+
+    @JsonProperty("datacenterAware")
+    public void setDatacenterAware(final DatacenterAware datacenterAware)
+    {
+        myDatacenterAware = datacenterAware;
+    }
+
+    @JsonProperty("datacenterAware")
+    public DatacenterAware getDatacenterAware()
+    {
+        return myDatacenterAware;
+    }
+
+    @JsonProperty("rackAware")
+    public void setRackAware(final RackAware rackAware)
+    {
+        myRackAware = rackAware;
+    }
+
+    @JsonProperty("rackAware")
+    public RackAware getRackAware()
+    {
+        return myRackAware;
+    }
+
+    @JsonProperty("hostAware")
+    public void setHostAware(final HostAware hostAware)
+    {
+        myHostAware = hostAware;
+    }
+
+    @JsonProperty("hostAware")
+    public HostAware getHostAware()
+    {
+        return myHostAware;
+    }
+
+    public enum ConnectionType
+    {
+        datacenterAware, rackAware, hostAware
+    }
+
+    public static final class DatacenterAware
+    {
+        private Map<String, Datacenter> myDatacenters = new HashMap<>();
+
+        public DatacenterAware()
+        {
+
+        }
+
+        @JsonProperty("datacenters")
+        public Map<String, Datacenter> getDatacenterAware()
+        {
+            return myDatacenters;
+        }
+
+        @JsonProperty("datacenters")
+        public void setDatacenters(final List<Datacenter> datacenters)
+        {
+            if (datacenters != null)
+            {
+                myDatacenters = datacenters.stream().collect(
+                        Collectors.toMap(Datacenter::getName, ks -> ks));
+            }
+        }
+
+        public static final class Datacenter
+        {
+            private String myName = "datacenter1";
+
+            public Datacenter()
+            {
+
+            }
+
+            public Datacenter(final String name)
+            {
+                myName = name;
+            }
+
+            @JsonProperty("name")
+            public String getName()
+            {
+                return myName;
+            }
+
+            @JsonProperty("name")
+            public void setName(final String name)
+            {
+                myName = name;
+            }
+        }
+    }
+
+    public static final class RackAware
+    {
+        private Map<String, Rack> myRackAware = new HashMap<>();
+
+        public RackAware()
+        {
+
+        }
+
+        @JsonProperty("racks")
+        public Map<String, Rack> getRackAware()
+        {
+            return myRackAware;
+        }
+
+        @JsonProperty("racks")
+        public void setMyRackAware(final List<Rack> rackAware)
+        {
+            if (rackAware != null)
+            {
+                myRackAware = rackAware.stream().collect(
+                        Collectors.toMap(Rack::getRackName, ks -> ks));
+            }
+        }
+
+        public static final class Rack
+        {
+            private String myDatacenterName = "datacenter1";
```

Comment:
I'm not sure we should have default names for datacenters/racks, unless perhaps cassandra itself provides similar defaults.